src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/auth/register/DetachedRegisterPage.java

```diff
@@ -0,0 +1,75 @@
+package com.wikia.webdriver.pageobjectsfactory.pageobject.auth.register;
+
+import com.wikia.webdriver.pageobjectsfactory.componentobject.modalwindows.FacebookSignupModalComponentObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.DetachedWindow;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.FacebookAuthContext;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.signin.SignInPage;
+
+public class DetachedRegisterPage extends DetachedWindow implements RegisterPage,
+  FacebookAuthContext {
+
+  private AttachedRegisterPage registerPage;
+  private String title = RegisterPage.pageTitle;
+
+  public DetachedRegisterPage() {
+    registerPage = new AttachedRegisterPage();
+  }
+
+  public DetachedRegisterPage(AttachedRegisterPage registerPage) {
+    registerPage = registerPage;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "registerPage". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)